src/msgraph_core/models/page_result.py

```diff
@@ -0,0 +1,127 @@
+"""
+This module defines the PageResult class which represents a page of
+items in a paged response.
+
+The PageResult class provides methods to get and set the next link and
+the items in the page, create a PageResult from a discriminator value, set
+the value, get the field deserializers, and serialize the PageResult.
+
+Classes:
+    PageResult: Represents a page of items in a paged response.
+"""
+from typing import Any, List, Optional
+
+from kiota_abstractions.serialization.parsable import Parsable  # type: ignore
+from kiota_abstractions.serialization.serialization_writer \
+     import SerializationWriter  # type: ignore
+from kiota_abstractions.serialization.parse_node import ParseNode  # type: ignore
+from typing import TypeVar, List, Optional, Generic
+
+T = TypeVar('T')
+
+
+class PageResult(Parsable, Generic[T]):
+    """
+    Represents a page of items in a paged response.
+    """
+    object_type = None
+
+    def __init__(self, object_type: Optional[Any] = None) -> None:
+        PageResult.object_type = object_type
+        self._odata_next_link: Optional[str] = None
+        self._value: Optional[List[T]] = None
+
+    @property
+    def odata_next_link(self) -> Optional[str]:
+        """
+        Gets the next link for the page.
+        Returns:
+            Optional[str]: The next link, or None if there is no next link.
+        """
+        return self._odata_next_link
+
+    @odata_next_link.setter
+    def odata_next_link(self, next_link: Optional[str]) -> None:
+        """
+        Sets the next link for the page.
+        Args:
+            next_link (Optional[str]): The next link to set.
+        """
+        self._odata_next_link = next_link
+
+    @property
+    def value(self) -> Optional[List[T]]:
+        """
+        Gets the items in the page.
+        Returns:
+            Optional[List[Any]]: The items in the page, or None if there
+            are no items.
+        """
+        return self._value
+
+    @value.setter
+    def value(self, value: Optional[List[T]]) -> None:
+        """
+        Sets the items in the page.
+        Args:
+            value (Optional[List[Any]]): The items to set.
+        """
+        self._value = value
+
+    @staticmethod
+    def create_from_discriminator_value(parse_node: ParseNode) -> 'PageResult':  # pylint: disable=unused-argument
```

Comment:
[suggestion] To avoid using 'PageResult' which affects type hinting, use `from __future__ import annotations` to enable postponed evaluation of annotations.